internal/pkg/manifest/validate.go

```diff
@@ -1609,14 +1638,17 @@ func validateTargetContainer(opts validateTargetContainerOpts) error {
 	}
 	targetContainer := aws.StringValue(opts.targetContainer)
 	if targetContainer == opts.mainContainerName {
+		if opts.mainContainerPort == nil {
+			return fmt.Errorf("target container %q doesn't expose any port", targetContainer)
+		}
 		return nil
 	}
 	sidecar, ok := opts.sidecarConfig[targetContainer]
 	if !ok {
-		return fmt.Errorf("target container %s doesn't exist", targetContainer)
+		return fmt.Errorf("target container %q doesn't exist", targetContainer)
 	}
 	if sidecar.Port == nil {
-		return fmt.Errorf("target container %s doesn't expose any port", targetContainer)
+		return fmt.Errorf("target container %q doesn't expose any port", targetContainer)
```

Comment:
probably more accurate as we only support 1 (same above)
```suggestion
		return fmt.Errorf("target container %q doesn't expose a port", targetContainer)
```